### ISSUE TITLE
Unable to set --version to a commit hash

### ISSUE THREAD

**Issue by JonZeolla** [zeek/package-manager#46]
As far as I'm aware, anything that can be checked out should be able to be used as the `--version`, but when I attempt to use the following commit hash that doesn't seem to work.

```
[root@715e1db6516e ~]# bro-pkg -vvv install bro/apache/metron-bro-plugin-kafka --version 279a2970b6a7b8fd7d91036b0b07798d03dbda71
2019-03-21 17:08:37 DEBUG    init Manager version 1.5.2
2019-03-21 17:08:37 DEBUG    found source clone of "bro" at /root/.bro-pkg/clones/source/bro
2019-03-21 17:08:38 DEBUG    getting info on "bro/apache/metron-bro-plugin-kafka"
error: invalid package "bro/apache/metron-bro-plugin-kafka": no such commit, branch, or version tag: "279a2970b6a7b8fd7d91036b0b07798d03dbda71"
[root@715e1db6516e ~]# git clone git://github.com/apache/metron-bro-plugin-kafka
Cloning into 'metron-bro-plugin-kafka'...
remote: Enumerating objects: 19, done.
remote: Counting objects: 100% (19/19), done.
remote: Compressing objects: 100% (15/15), done.
remote: Total 332 (delta 5), reused 12 (delta 4), pack-reused 313
Receiving objects: 100% (332/332), 91.22 KiB | 0 bytes/s, done.
Resolving deltas: 100% (193/193), done.
[root@715e1db6516e ~]# cd metron-bro-plugin-kafka/
[root@715e1db6516e metron-bro-plugin-kafka]# git checkout 279a2970b6a7b8fd7d91036b0b07798d03dbda71
Note: checking out '279a2970b6a7b8fd7d91036b0b07798d03dbda71'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at 279a297... METRON-1865 Fix metron-bro-plugin-kafka tests (JonZeolla via jonzeolla) closes apache/metron-bro-plugin-kafka#16
[root@715e1db6516e metron-bro-plugin-kafka]# git log -1
commit 279a2970b6a7b8fd7d91036b0b07798d03dbda71
Author: JonZeolla <zeolla@gmail.com>
Date:   Fri Nov 9 20:17:12 2018 -0500

    METRON-1865 Fix metron-bro-plugin-kafka tests (JonZeolla via jonzeolla) closes apache/metron-bro-plugin-kafka#16
```

Here's where that error message is coming from:
https://github.com/zeek/package-manager/blob/3505a6b11e0d3872dbd591bed21bf545b45ec292/bropkg/manager.py#L1112-L1143

After a quick skim, I wonder if the shource is here:
https://github.com/zeek/package-manager/blob/3505a6b11e0d3872dbd591bed21bf545b45ec292/bropkg/manager.py#L2160-L2175

I didn't look into what `clone.tags` looks like though.

**Comment by jsiwek** [comment]
> [root@715e1db6516e ~]# bro-pkg -vvv install bro/apache/metron-bro-plugin-kafka --version 279a2970b6a7b8fd7d91036b0b07798d03dbda71
> 2019-03-21 17:08:37 DEBUG    init Manager version 1.5.2
> 2019-03-21 17:08:37 DEBUG    found source clone of "bro" at /root/.bro-pkg/clones/source/bro
> 2019-03-21 17:08:38 DEBUG    getting info on "bro/apache/metron-bro-plugin-kafka"
> error: invalid package "bro/apache/metron-bro-plugin-kafka": no such commit, branch, or version tag: "279a2970b6a7b8fd7d91036b0b07798d03dbda71"

Seems it may be re-using an existing/old clone located at `/root/.bro-pkg/clones/source/bro`, so you might check if the commit actually exists there (without doing a fetch).

If it doesn't exist, then likely you could do `bro-pkg refresh`, which would effectively `git fetch` and so that same `bro-pkg install --version` may work after that.

But then generally seems like the real fix may be that a `bro-pkg install` should automatically fetch in the case where we already had a local clone.

**Comment by jsiwek** [comment]
Oops, just noticed I read the log wrong.  The directory to check for pre-existing clone I think would be `/root/.bro-pkg/clones/package/metron-bro-plugin-kafka`.

**Comment by jsiwek** [comment]
Sorry, seems I can just repro and seems installing from a commit hash may just have never worked (or was broken when we optimized the git cloning process to do shallow clones).  It may happen to still work if that commit happens to be at the tip of some branch, but I think the only workaround for now may be using a branch. `bro-pkg` may ultimately need to be able to detect cases like this where a shallow clone is not sufficient.

**Comment by jsiwek** [comment]
I pushed a fix now in master or v1.5.4 if you'd like to test/confirm.

**Comment by JonZeolla** [comment]
LGTM, thanks
